piksi_tools/console/settings_view.py

```diff
@@ -568,29 +575,103 @@ def _settings_export_to_file_button_fired(self):
-            settings_out = {}
-            # iterate over nested dict and set inner value to a bare string rather than dict
-            for section in self.settings:
-                settings_out[section] = {}
-                for setting, inner_dict in self.settings[section].items():
-                    settings_out[section][setting] = str(inner_dict.value)
-            # write out with config parser
-            parser = configparser.RawConfigParser()
-            # the optionxform is needed to handle case sensitive settings
-            parser.optionxform = str
-            parser.read_dict(settings_out)
-            # write to ini file
-            try:
-                with open(file.path, "w") as f:
-                    parser.write(f)
-            except IOError as e:
-                print('Unable to export settings to file due to IOError: {}'.format(e))
-        else:  # No error message because user pressed cancel and didn't choose a file
-            pass
+
+        if file.open() != OK:
+            # No error message here because user likely pressed cancel when choosing file
+            return
+
+        print('Exporting settings to local path {0}'.format(file.path))
+        # copy settings so we can modify dict in place to write for configparser
+        settings_out = {}
+        # iterate over nested dict and set inner value to a bare string rather than dict
+        for section in self.settings:
+            settings_out[section] = {}
+            for setting, inner_dict in self.settings[section].iteritems():
+                settings_out[section][setting] = str(inner_dict.value)
+        # write out with config parser
+        parser = configparser.RawConfigParser()
+        # the optionxform is needed to handle case sensitive settings
+        parser.optionxform = str
+        parser.read_dict(settings_out)
+        # write to ini file
+        try:
+            with open(file.path, "w") as f:
+                parser.write(f)
+        except IOError as e:
+            print('Unable to export settings to file due to IOError: {}'.format(e))
+
+    def _import_success(self, count):
+        print("Successfully imported {} settings from file.".format(count))
+        confirm_prompt = prompt.CallbackPrompt(
+            title="Save to device flash?",
+            actions=[prompt.close_button, prompt.ok_button],
+            callback=self._settings_save_button_fired)
+        confirm_prompt.text = "\n" \
+            "  Settings import from file complete.  Click OK to save the settings  \n" \
+            "  to the device's persistent storage.  \n"
+        confirm_prompt.run(block=False)
+
+    def _import_fail(self):
+        confirm_prompt = prompt.CallbackPrompt(
+            title="Failed to import settings from file",
+            actions=[prompt.close_button])
+        confirm_prompt.text = "\n" \
+            "  Verify that config file is not overwriting active connection settings.  \n"
```

Comment:
Now that file importing is truly sending every setting from the file, I found it necessary to add this warning. @denniszollo

Reply:
That is a good warning.